core/jvm/src/main/scala/reftree/render/AnimatedGifRenderer.scala

```diff
@@ -6,17 +6,54 @@ import java.nio.charset.StandardCharsets
 import java.nio.file.Path
 import javax.xml.parsers.SAXParserFactory
 
-import com.sksamuel.scrimage.Image
+import com.sksamuel.scrimage.ImmutableImage
 import com.sksamuel.scrimage.nio.StreamingGifWriter
 import org.apache.batik.transcoder.{TranscoderInput, SVGAbstractTranscoder, TranscoderOutput}
 import org.apache.batik.transcoder.image.{ImageTranscoder, PNGTranscoder}
 import reftree.dot.Graph
 import reftree.svg.{OptimizedGraphAnimation, XmlSvgApi}
 import reftree.svg.animation.Frame
-
+import java.time.{Duration => JavaDuration}
+import java.time.temporal.ChronoUnit
+import java.util.concurrent.TimeUnit
 import scala.sys.process.{Process, BasicIO}
+import scala.concurrent.duration.{FiniteDuration, Duration => ScalaDuration}
 
 object AnimatedGifRenderer {
+
+  def toScala(duration: java.time.Duration): scala.concurrent.duration.FiniteDuration = {
+    val originalSeconds = duration.getSeconds
+    val originalNanos = duration.getNano
+    if (originalNanos == 0) {
+      if (originalSeconds == 0) ScalaDuration.Zero
+      else FiniteDuration(originalSeconds, TimeUnit.SECONDS)
+    } else if (originalSeconds == 0) {
+      FiniteDuration(originalNanos, TimeUnit.NANOSECONDS)
+    } else {
+      try {
+        val secondsAsNanos = Math.multiplyExact(originalSeconds, 1000000000)
+        val totalNanos = secondsAsNanos + originalNanos
+        if ((totalNanos < 0 && secondsAsNanos < 0) || (totalNanos > 0 && secondsAsNanos > 0)) FiniteDuration(totalNanos, TimeUnit.NANOSECONDS)
+        else throw new ArithmeticException()
+      } catch {
+        case _: ArithmeticException => throw new IllegalArgumentException(s"Java duration $duration cannot be expressed as a Scala duration")
+      }
+    }
+  }
+
+  def toJava(duration: scala.concurrent.duration.FiniteDuration): java.time.Duration = {
+    if (duration.length == 0) JavaDuration.ZERO
+    else duration.unit match {
+      case TimeUnit.NANOSECONDS => JavaDuration.ofNanos(duration.length)
+      case TimeUnit.MICROSECONDS => JavaDuration.of(duration.length, ChronoUnit.MICROS)
+      case TimeUnit.MILLISECONDS => JavaDuration.ofMillis(duration.length)
+      case TimeUnit.SECONDS => JavaDuration.ofSeconds(duration.length)
+      case TimeUnit.MINUTES => JavaDuration.ofMinutes(duration.length)
+      case TimeUnit.HOURS => JavaDuration.ofHours(duration.length)
+      case TimeUnit.DAYS => JavaDuration.ofDays(duration.length)
+    }
+  }
+
```

Comment:
I copied this from https://github.com/scala/scala-java8-compat/blob/main/src/main/scala/scala/compat/java8/DurationConverters.scala. I don't want to include another dependency.

Reply:
I am thinking, this update is bringing some new major versions of dependencies. So we are already potentially breaking compatibility for some downstream code. Why not just switch to the Java `Duration` and bump the major version of `reftree`?

Reply:
okay I will switch to java Duration.

Reply:
updated to use java duration